core/extension/gdextension_interface.cpp

```diff
@@ -1194,6 +1194,33 @@ static GDObjectInstanceID gdextension_object_get_instance_id(GDExtensionConstObj
 	return (GDObjectInstanceID)o->get_instance_id();
 }
 
+static GDExtensionBool gdextension_object_has_script_method(GDExtensionConstObjectPtr p_object, GDExtensionConstStringNamePtr p_method) {
+	Object *o = (Object *)p_object;
+	const StringName method = *reinterpret_cast<const StringName *>(p_method);
+
+	ScriptInstance *script_instance = o->get_script_instance();
+	if (script_instance) {
+		return script_instance->has_method(method);
+	}
+	return false;
+}
+
+static void gdextension_object_call_script_method(GDExtensionObjectPtr p_object, GDExtensionConstStringNamePtr p_method, const GDExtensionConstVariantPtr *p_args, GDExtensionInt p_argument_count, GDExtensionUninitializedVariantPtr r_return, GDExtensionCallError *r_error) {
+	Object *o = (Object *)p_object;
+	const StringName method = *reinterpret_cast<const StringName *>(p_method);
+	const Variant **args = (const Variant **)p_args;
```

Comment:
I know this happens a lot in existing code, but is there any guideline regarding when to choose `reinterpret_cast` and when C-style casts?

Reply:
Unfortunately, I don't think we have any guidelines. This looks as it does because I copy-pasted it from `gdextension_variant_call()`. :-) I could change the 2nd cast to `reinterpret_cast`, but that doesn't really improve the overall situation. We could save fixing this for an overhaul of all the casts used in GDExtension, where we could establish some sort of guidelines?

Reply:
Yes, let's evaluate it outside this PR then 🙂

Reply:
I vote for addressing the whole casting situation in a dedicated PR, and leaving it out-of-scope for this PR. The changes here match what we've already got.